go/libraries/doltcore/table/untyped/tabular/fixedwidth_tablewriter.go

```diff
@@ -142,28 +164,30 @@ func (w *FixedWidthTableWriter) WriteColoredRow(ctx context.Context, r sql.Row,
 	return nil
 }
 
-func (w *FixedWidthTableWriter) sampleRow(r sql.Row, colors []*color.Color) (tableRow, error) {
+func (w *FixedWidthTableWriter) sampleRow(r []string, widths []FixedWidthString, colors []*color.Color) (tableRow, error) {
+	if len(widths) == 0 {
+		widths = make([]FixedWidthString, len(r))
+	}
 	row := tableRow{
-		columns: make([]string, len(r)),
+		columns: r,
```

Comment:
Kinda weird that we have both `columns` and `widths` here... seems like this separation is a holdover from the previous implementation that use integer widths. FixedWidthString is the processed version of the string result and should be self-contained, usable for printing the table by itself. No need for a separate set of unprocessed column strings, right?